### PR TITLE
Refactored js-country-picker-select into stimulus controller and util…

### PR DESCRIPTION
**Pull request by HWallenberg**
### Jira link

https://transformuk.atlassian.net/browse/OTT-351
https://transformuk.atlassian.net/browse/OTT-373

### What?

I have added/removed/altered:

- [x] Added Country_Select_Box Stimulus controller, Utility Class and relevant specs
- [x] Removed redundant js-country-picker-select code from commodities.js
- [x] Altered Trading Partner Model to valid blank 'All Countries' selection.
- [x] Altered Country_Picker view to bind to new stimulus controller 

### Why?

I am doing this because:

- Task was to refactor js-country-picker-select into stimulus controller and utility class as part of https://transformuk.atlassian.net/browse/OTT-351 and write appropriate unit tests.
-  Updated trading_partner model to validate blanks as 'all countries' when javascript disabled.

### Have you? (optional)

- [x] Reviewed view changes with stake holders
- [x] Validated mobile responsive behaviour of view changes

### CODE REVIEW

File: app/webpacker/controllers/country_select_box_controller.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    const autocompleteInput = this.countrySelectTarget.querySelector('input.autocomplete__input');
+
+    if (autocompleteInput) {
+      autocompleteInput.addEventListener('focus', this.clearSelect.bind(this));
+    } else {
+      console.error('Autocomplete input element not found.');
+    }
+  }
+
+  disconnect() {
```
Comment (willfish) [inline app/webpacker/controllers/country_select_box_controller.js]: How do you feel about putting the public methods (e.g. those that don't have a # prefix above the private methods?

File: app/webpacker/src/javascripts/commodities.js [hunk 2]
```diff
@@ -527,51 +527,6 @@
       */
     selectBoxes: {
       initialize: function() {
-        $('.js-select').each(function() {
```
Comment (willfish) [inline app/webpacker/src/javascripts/commodities.js:530]: Yay \0/

File: spec/javascript/controllers/country_select_box_controller_spec.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('clears the select value on focus', () => {
+    const element = document.querySelector('[data-controller="country-select-box"]');
+    const inputElement = element.querySelector('.autocomplete__input');
+
+    inputElement.value = 'All countries';
+    const event = new Event('focus');
+    inputElement.dispatchEvent(event);
+
+    expect(inputElement.value).toEqual('');
```
Comment (willfish) [inline spec/javascript/controllers/country_select_box_controller_spec.js:39]: Lovely jubbly

**Review by willfish — DISMISSED** [review]
This is beautiful

File: app/webpacker/controllers/country_select_box_controller.js (hunk 1 above)
Comment (HWallenberg) [reply app/webpacker/controllers/country_select_box_controller.js]: yep should be, done.

**Review by willfish — APPROVED** [review]

**Pull request merged**